libs/react/src/lib/form/input/input.tsx

```diff
@@ -65,15 +66,13 @@ export const Input = ({
     setLocalValue(value)
   }, [value])
 
-  const localOnChange = useCallback(
+  const handleChange = useCallback(
     (event: ChangeEvent<HTMLInputElement>) => {
       const oldValue = event.target.value
       const newValue = formatter ? formatter(oldValue) : oldValue
       setLocalValue(newValue)
 
-      event.target.value = newValue
-
-      if (onChange) onChange(event)
+      if (onChange) onChange({ ...event, target: { ...event.target, value: newValue } });
```

Comment:
This is the only way I got formatting to work so that the caret doesn't jump to the end when removing text in the middle of the field. For some reason doing event.target.value = newValue does not work.